streams/src/main/java/org/apache/kafka/streams/kstream/internals/SessionWindowedCogroupedKStreamImpl.java

```diff
@@ -0,0 +1,156 @@
+    private final SessionWindows sessionWindows;
+    private final CogroupedStreamAggregateBuilder<K, V> aggregateBuilder;
+    private final Map<KGroupedStreamImpl<K, ?>, Aggregator<? super K, ? super Object, V>> groupPatterns;
+
+    SessionWindowedCogroupedKStreamImpl(final SessionWindows sessionWindows,
+                                        final InternalStreamsBuilder builder,
+                                        final Set<String> sourceNodes,
+                                        final String name,
+                                        final Serde<K> keySerde,
+                                        final Serde<V> valSerde,
+                                        final CogroupedStreamAggregateBuilder<K, V> aggregateBuilder,
+                                        final StreamsGraphNode streamsGraphNode,
+                                        final Map<KGroupedStreamImpl<K, ?>, Aggregator<? super K, ? super Object, V>> groupPatterns) {
+        super(name, null, null, sourceNodes, streamsGraphNode, builder);
+        //keySerde and valueSerde are null because there are many different groupStreams that they could be from
+        this.sessionWindows = sessionWindows;
+        this.aggregateBuilder = aggregateBuilder;
+        this.groupPatterns = groupPatterns;
+    }
+
+    @Override
+    public KTable<Windowed<K>, V> aggregate(final Initializer<V> initializer,
+                                            final Merger<? super K, V> sessionMerger) {
+        return aggregate(initializer, sessionMerger, Materialized.with(keySerde, null));
+    }
+
+    @Override
+    public KTable<Windowed<K>, V> aggregate(final Initializer<V> initializer,
+                                            final Merger<? super K, V> sessionMerger,
+                                            final Materialized<K, V, SessionStore<Bytes, byte[]>> materialized) {
+        return aggregate(initializer, NamedInternal.empty(), sessionMerger, materialized);
+    }
+
+    @Override
+    public KTable<Windowed<K>, V> aggregate(final Initializer<V> initializer,
+                                            final Named named,
+                                            final Merger<? super K, V> sessionMerger) {
+        return aggregate(initializer, named, sessionMerger, Materialized.with(keySerde, null));
+    }
+
+    @Override
+    public KTable<Windowed<K>, V> aggregate(final Initializer<V> initializer,
+                                            final Named named,
+                                            final Merger<? super K, V> sessionMerger,
+                                            final Materialized<K, V, SessionStore<Bytes, byte[]>> materialized) {
+        Objects.requireNonNull(initializer, "initializer can't be null");
+        Objects.requireNonNull(sessionMerger, "sessionMerger can't be null");
+        Objects.requireNonNull(materialized, "materialized can't be null");
+        Objects.requireNonNull(named, "named can't be null");
```

Comment:
@bbejeck -- I was double checking existing code and many (all) operator actually allow `named` to be `null` -- do you remember why? I am wondering if this is a issue with existing code or an issue with the new code?

Reply:
Since there are options to just not pass it in anyone who would put null should just not include it

Reply:
>@bbejeck -- I was double checking existing code and many (all) operator actually allow named to be null -- do you remember why? I am wondering if this is a issue with existing code or an issue with the new code?

Which operators are you referring to? I just took a quick look at `KStreamImpl` and they don't allow null `Named`.  Several of the overloads don't take a `Named` and create a `NamedInternal#empty`

Reply:
There are a couple of examples, but your comment indicates, that they should allow it. Just wanted to confirm. Will go over the code and add the corresponding checks in a follow up PR to fix it.